databuilder/query_engines/base_sql.py

```diff
@@ -251,7 +251,7 @@ def get_sql_string_contains(self, node):
 
     def string_replace(self, value, pattern, replacement):
         return SQLFunction(
-            "REPLACE", value, pattern, replacement, type=sqlalchemy_types.String
+            "REPLACE", value, pattern, replacement, type_=sqlalchemy_types.String
```

Comment:
Not exactly. There are quite a few cases (of which this is one) where it doesn't really matter what type SQLAlchemy thinks a value is because it generates identical SQL in any case. So in a sense, the type here is pointless. But I don't want to have to think about where types are and aren't needed and so I think wherever we define a custom function we should set its return type appropriately.